### PR TITLE
feat(ui): optimize header

### PR DESCRIPTION
**Pull request by JulienMassonnat**
A PR to optimize several things on the header : 

- correct overlapping effect of the dropdown menus
- Add orange color when we are hover the menu item 
- Remove the active border of the toggle button
- Transform the burger menu icon into a close button when active

## Summary by CodeRabbit

- **New Features**
	- Introduced an expandable menu in the header bar, enhancing navigation usability.
	- Added a new `DropdownTitle` component for a unified presentation of dropdown titles with icons.
- **Style**
	- Updated the header bar and dropdown styles for better aesthetics and responsiveness.
- **Bug Fixes**
	- Fixed an issue with dropdown items not displaying correctly on various screen sizes.

### CODE REVIEW

File: app/components/headerBar.tsx [hunk 1]
```diff
@@ -8,17 +9,49 @@ import SocialIcons from "app/components/socialIcons";
...
+
   return (
-    <Navbar fixed="top" collapseOnSelect expand="xl" bg="white">
+    <Navbar fixed="top" collapseOnSelect expand="xl" bg="white" expanded={expanded}>
       <Container>
         <Navbar.Brand href={internalUrls.home}>
           <AxoneLogo width="200" />
         </Navbar.Brand>
-        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
+        <Navbar.Toggle aria-controls="responsive-navbar-nav" onClick={() => setExpanded(expanded ? false : 'expanded')}>
```
Comment (ccamel) [inline app/components/headerBar.tsx]: `setExpanded` function expect a boolean as argument and not a boolean union a string value. Why not:

```suggestion
        <Navbar.Toggle aria-controls="responsive-navbar-nav" onClick={() => setExpanded(!expanded)}>
```

File: app/components/headerBar.tsx [hunk 2]
```diff
@@ -8,17 +9,49 @@ import SocialIcons from "app/components/socialIcons";
...
+          </svg>
+          )}
+        </Navbar.Toggle>
         <Navbar.Collapse id="responsive-navbar-nav">
           <Nav className="me-auto" />
           <Nav className="xl:space-x-10">
-            <NavDropdown title="LEARN" className="text-xl font-bold">
+          <NavDropdown
+            title={
+    <span className="arrow-dropdown">
```
Comment (ccamel) [inline app/components/headerBar.tsx]: AFAIS, this span is duplicated and used for LEARN and BUILD. For DRY consideration, you could create a reusable component for this dropdown title. And use it where necessary. For instance:

```typescript
const DropdownTitle = ({ title }: DropdownTitleProps) => (
  <span className="arrow-dropdown">
    {title}
    <svg ...
    >
...
     </svg>
  </span>
);
```

File: app/styles/headerBar.css [hunk 3]
```diff
@@ -28,6 +59,9 @@
   .socialIcons {
     height: 78px;
   }
+  .nav-link svg {
+    margin-left: 2px; /* Encore plus proche pour les petits écrans */
```
Comment (ccamel) [inline app/styles/headerBar.css]: Remove unnecessary comments (especially if they are not in English). Here, I believe the css rule is self-explanatory.

**Review by ccamel — CHANGES_REQUESTED** [review]
Thanks for the enhancements! There are a few recommended changes I've outlined that need to be taken into account before we proceed with the merge.

File: app/components/headerBar.tsx (hunk 2 above)
Comment (JulienMassonnat) [reply app/components/headerBar.tsx]: Thanks ! Much better indeed !

**Review by JulienMassonnat — COMMENTED** [review]
Ok thanks I did a new commit

**Review by ccamel — APPROVED** [review]
Good! Thanks :+1:

**Pull request merged**